pytket/qir/conversion/conversion.py

```diff
@@ -493,50 +493,98 @@ def circuit_to_module(
+                        conditional sub-circuit when the condition is True.
+                        """
+                        if op.value == 1:
+                            self.circuit_to_module(conditional_circuit, module)
+
+                    def condition_block_false() -> None:
+                        """
+                        Populate recursively the module with the contents of the
+                        conditional sub-circuit when the condition is False.
+                        """
+                        if op.value == 0:
+                            self.circuit_to_module(conditional_circuit, module)
+
+                    if condition_name in self.ssa_vars:
+
+                        ssabool = module.builder.call(
+                            self.read_bit_from_reg,
+                            [
+                                self.ssa_vars[condition_name],
+                                pyqir.const(self.qir_int_type, condition_bit_index),
+                            ],
+                        )
 
-                def condition_block_false() -> None:
-                    """
-                    Populate recursively the module with the contents of the conditional
-                    sub-circuit when the condition is False.
-                    """
-                    if op.value == 0:
-                        self.circuit_to_module(conditional_circuit, module)
+                        module.module.builder.if_(
+                            ssabool,
+                            true=lambda: condition_block_true(),  # type: ignore
+                            false=lambda: condition_block_false(),  # type: ignore
+                        )
 
-                if condition_name in self.ssa_vars:
+                    else:
+                        ValueError(
+                            """circuit contruction with this condition
+                            not yet supported"""
+                        )
+                        # this should be an assertion when working
+                else:
+                    conditional_circuit = self._rebase_op_to_gateset(
+                        op.op, command.args[op.width :]
+                    )
+                    condition_name = command.args[0].reg_name
 
-                    ssabool = module.builder.call(
-                        self.read_bit_from_reg,
-                        [
-                            self.ssa_vars[condition_name],
-                            pyqir.const(self.qir_int_type, condition_bit_index),
-                        ],
+                    for i in range(op.width):
+                        if command.args[i].reg_name != condition_name:
+                            raise ValueError(
+                                "conditional can only work with one entire register"
+                            )
+
+                    for i in range(op.width - 1):
+                        if command.args[i].index[0] >= command.args[i + 1].index[0]:
```

Comment:
Do we need to insist that the bit indices are monotonically increasing? We should be able to handle e.g. `condition_bits=[1,0], condition_value=2` which is equivalent to `condition_bits=[0,1], condition_value=1` (just needs some bit twiddiling on the value).

Reply:
I have copied this checks from the qasm conversion. Do we want to support things that have not been possible in qasm? It would be possible as well to do subsets of a register as well if we want to? I have just skipped that for the same reason.

Reply:
OK, if this is a restriction of the QASM converter it's probably fine for now.